pandas/core/algorithms.py

```diff
@@ -1691,6 +1693,14 @@ def map_array(
         if na_action == "ignore":
             mapper = mapper[mapper.index.notna()]
 
+        if isinstance(arr.dtype, ArrowDtype) and arr.dtype.name.startswith("timestamp"):
+            try:
+                # Convert elements to pandas.Timestamp (or datetime64[ns])
+                arr = arr.astype("datetime64[ns]")
+            except Exception:
```

Comment:
This is the wrong place to fix this. This should be fixed in `ArrowExtensionArray.map`